crates/hyperqueue/src/server/state.rs

```diff
@@ -263,7 +268,12 @@ impl State {
 
         for task_id in msg.running_tasks {
             let job = self.get_job_mut_by_tako_task_id(task_id).unwrap();
-            job.set_waiting_state(task_id);
+            job.increment_task_crash_count(task_id);
+            if job.is_often_crashing_task(task_id) {
+                cancel_tasks_from_callback(state_ref, tako_ref, job.job_id, vec![task_id]);
```

Comment:
For Tako scheduler it is better if task are cancelled atomically at once; hence I propose collecting task_ids for cancel first and then (if non-empty) trigger cancel_tasks_from_callback only once. But this is only a minor issue.